tests/unit/synapseutils/unit_test_synapseutils_describe.py

```diff
@@ -0,0 +1,63 @@
+import pytest
+import synapseclient
+import pandas as pd
+import tempfile
+from synapseutils.describe import _open_entity_as_df, synapse_describe
+from unittest.mock import patch, MagicMock
+
+
+class TestOpenEntityAsDf:
+
+    id = 'syn123456'
+    df = pd.DataFrame(
+            {'gene': ['MSN', 'CD44', 'MSN', 'CD44', 'MSN', 'CD44', 'MSN', 'CD44', 'CD44'],
+              "score": [1, 2, 1, 2, 1, 2, 1, 2, 1]
+             })
+
+    csv_path = tempfile.NamedTemporaryFile(delete=False, suffix='.csv')
+    tsv_path = tempfile.NamedTemporaryFile(delete=False, suffix='.tsv')
+
+    @pytest.fixture(autouse=True, scope='function')
+    def init_syn(self, syn):
+        self.syn = syn
+
+    @pytest.fixture
+    def setup_csv(self):
+        self.df.to_csv(self.csv_path)
+        self.csv_ent = synapseclient.File(name='file.csv',
+                                          id=self.id,
+                                          path=self.csv_path.name,
+                                          parentId='8765')
+
+    @pytest.fixture
+    def setup_tsv(self):
+        self.df.to_csv(self.tsv_path, sep='\t')
+        self.tsv_ent = synapseclient.File(name='file.tsv',
+                                          id=self.id,
+                                          path=self.tsv_path.name,
+                                          parentId='8765')
+
+    def test_open_entity_as_df_csv(self, setup_csv):
+        with patch.object(self.syn, "get", return_value=self.csv_ent) as mocked_get_entity:
+
+            assert type(_open_entity_as_df(self.syn, self.id)) == pd.DataFrame
```

Comment:
Instead of just asserting the type, you can also assert that the data frame returned is what you expect it to be.   I think pandas has some: https://pandas.pydata.org/docs/reference/api/pandas.testing.assert_frame_equal.html

Reply:
I added some tests for the values here.  Unfortunately here, I had to perform individual checks and not compare the dataframes because I return the dtype of the column as a value.  It was turning into a time sink, but if you know where dtype('O') comes from so I can create a dataframe with it, let me know, and I'll just convert the individual tests to a single assert_dataframe_equals!